src/gateway/types.js

```diff
@@ -91,14 +92,16 @@ export interface AbortablePromise<T> extends Promise<T> {
  */
 export type GetHeaderCallback = (name: string) => ?string;
 
-/**
- * Callback to begin a trace session.
- *
- * @param {string} name The name of the traced action.
- * @returns {ITraceSession} A trace session that the caller should use to
- * indicate when the session is finished.
- */
-export type TraceCallback = (name: string) => ITraceSession;
+export interface TraceCallback {
```

Comment:
I was toying with getting better editor integration for function calls not typed inline at their usage points.